controllers/storagecluster/external_resources.go

```diff
@@ -380,6 +381,9 @@ func (r *StorageClusterReconciler) createExternalStorageClusterResources(instanc
 				scc = newCephBlockPoolStorageClassConfiguration(instance)
 				// update the storageclass name to rados storagesclass name
 				scc.storageClass.Name = fmt.Sprintf("%s-%s", instance.Name, d.Name)
+			} else if d.Name == cephRbdTopologyStorageClassName && instance.Spec.ManagedResources.CephNonResilientPools.Enable {
+				scc = newNonResilientCephBlockPoolStorageClassConfiguration(instance)
```

Comment:
will the replicated pool will be also set, which is passed by "pool" in json

Reply:
Yes as of now that field will be set as AFAIK that's a required field. You were working on some changes to remove that field as a requirement when topologyConstrainedPools are specified, do let me know the progress of that and we can accordingly decide.

Reply:
yes correct but currently we need it to set